GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m

```diff
@@ -1087,7 +1087,135 @@ - (void)testApplicationDidReceiveRemoteNotificationWithCompletionIsInvokedOnInte
+    XCTAssertNotNil(localCompletionHandler);
+    localCompletionHandler(UIBackgroundFetchResultFailed);
+  };
+
+  id interceptor = OCMProtocolMock(@protocol(GULApplicationDelegate));
+  OCMExpect([interceptor application:application
+                didReceiveRemoteNotification:notification
+                      fetchCompletionHandler:[OCMArg isNotNil]])
+      .andDo(onDidReceiveRemoteNotification);
+
+  GULTestAppDelegate *testAppDelegate = [[GULTestAppDelegate alloc] init];
+  OCMStub([self.mockSharedApplication delegate]).andReturn(testAppDelegate);
+  [GULAppDelegateSwizzler proxyOriginalDelegateIncludingAPNSMethods];
+
+  [GULAppDelegateSwizzler registerAppDelegateInterceptor:interceptor];
+
+  [testAppDelegate application:application
+      didReceiveRemoteNotification:notification
+            fetchCompletionHandler:completion];
+  testAppDelegate.remoteNotificationCompletionHandler(UIBackgroundFetchResultFailed);
+  OCMVerifyAll(interceptor);
+  [self waitForExpectations:@[ completionExpectation ] timeout:0.1];
+}
+
+- (void)
+    testApplicationDidReceiveRemoteNotificationWithCompletionCompletionIsCalledOnce_HandleNewDataState {
+  NSDictionary *notification = @{};
+  GULApplication *application = [GULApplication sharedApplication];
+
+  XCTestExpectation *completionExpectation =
+      [[XCTestExpectation alloc] initWithDescription:@"Completion called once"];
+  completionExpectation.assertForOverFulfill = YES;
+
+  void (^completion)(UIBackgroundFetchResult) = ^(UIBackgroundFetchResult result) {
+    XCTAssertEqual(result, UIBackgroundFetchResultNewData);
+    [completionExpectation fulfill];
+  };
+
+  void (^onDidReceiveRemoteNotification)(NSInvocation *invocation) = ^(NSInvocation *invocation) {
+    void __unsafe_unretained (^localCompletionHandler)(UIBackgroundFetchResult) = nil;
+    [invocation getArgument:(void *)(&localCompletionHandler) atIndex:4];
+    XCTAssertNotNil(localCompletionHandler);
+    localCompletionHandler(UIBackgroundFetchResultFailed);
+  };
+
+  id interceptor = OCMProtocolMock(@protocol(GULApplicationDelegate));
+  OCMExpect([interceptor application:application
+                didReceiveRemoteNotification:notification
+                      fetchCompletionHandler:[OCMArg isNotNil]])
+      .andDo(onDidReceiveRemoteNotification);
+
+  GULTestAppDelegate *testAppDelegate = [[GULTestAppDelegate alloc] init];
+  OCMStub([self.mockSharedApplication delegate]).andReturn(testAppDelegate);
+  [GULAppDelegateSwizzler proxyOriginalDelegateIncludingAPNSMethods];
+
+  [GULAppDelegateSwizzler registerAppDelegateInterceptor:interceptor];
+
+  [testAppDelegate application:application
+      didReceiveRemoteNotification:notification
+            fetchCompletionHandler:completion];
+  testAppDelegate.remoteNotificationCompletionHandler(UIBackgroundFetchResultNewData);
+  OCMVerifyAll(interceptor);
+  [self waitForExpectations:@[ completionExpectation ] timeout:0.1];
```

Comment:
nit: Looks like these 3 tests are mostly identical except the background fetch result. Could you please extract the test logic to a helper function like `assertApplicationDidReceiveRemoteNotificationWithCompletionCompletionIsCalledOnceWithResult:(UIBackgroundFetchResult)` to reduce code duplication.

Reply:
_I thought it's paid by the number of lines_